lib/Cake/View/Helper/FormHelper.php

```diff
@@ -379,6 +379,13 @@ public function create($model = null, $options = array()) {
 		$this->inputDefaults($options['inputDefaults']);
 		unset($options['inputDefaults']);
 
+		if (isset($options['action'])) {
+			trigger_error('Using key `action` is deprecated, use `url` directly instead.', E_USER_DEPRECATED);
+		}
+		if (isset($options['url']) && isset($options['url']['action'])) {
+			$options['action'] = $options['url']['action'];
+		}
```

Comment:
It is necessary for the internals of this method to keep the ID generation intact.